packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -284,17 +284,40 @@ class EngineBuildPaths {
   final String? webSdk;
 }
 
-/// Information about a local engine build
+/// Information about a local engine build (i.e. `--local-engine[-host]=...`).
+///
+/// See https://github.com/flutter/flutter/wiki/The-flutter-tool#using-a-locally-built-engine-with-the-flutter-tool
+/// for more information about local engine builds.
 class LocalEngineInfo {
+  /// Creates a reference to a local engine build.
+  ///
+  /// The [engineOutPath] and [engineHostOutPath] are assumed to be resolvable
+  /// paths to the built engine artifacts for the target (device) and host
+  /// (build) platforms, respectively.
   const LocalEngineInfo({
     required this.engineOutPath,
     required this.engineHostOutPath,
-    required this.localEngineName,
   });
 
+  /// The path to the engine artifacts for the target (device) platform.
+  ///
+  /// For example, if the target platform is Android debug, this would be a path
+  /// like `/path/to/engine/src/out/android_debug_unopt`. To retrieve just the
+  /// name (platform), see [localEngineName].
   final String engineOutPath;
```

Comment:
Unrelated, but I wonder if we should call this targetEnginePath for clarity. Technically these are both engine out paths.

Reply:
I'm supportive, done and uploaded!